### PR TITLE
Feature/bmc 18922

### PR DESCRIPTION
**Pull request by yandrickg**

### CODE REVIEW

File: .github/workflows/validate-all.yml [hunk 1]
```diff
@@ -47,6 +47,16 @@ jobs:
     with:
       module-name: LocationApi
       test-file: tc_location_api.rb
+  build-and-test-paymentsapi:
```
Comment (fybJenkins) [inline .github/workflows/validate-all.yml:50]: was `build-and-test-billingapi` forgotten here?

File: tests/tc_billing_api.rb [hunk 2]
```diff
@@ -0,0 +1,284 @@
...
+    self.verify_called_once expectation
+  end
+
+  def test_get_product_availability
+    request, response = TestUtils.generate_payloads_from('billingapi/products/product_availability_get')
+    expectation = TestUtils.setup_expectation(request, response, 1)
+
+    api_instance = BillingApi::ProductsApi.new
+    opts = {
+      product_category: ['SERVER'],
```
Comment (fybJenkins) [inline tests/tc_billing_api.rb:60]: Why not use `TestUtils.generate_query_params` here?

File: tests/tc_invoicing_api.rb [hunk 3]
```diff
@@ -0,0 +1,115 @@
...
+    self.verify_called_once expectation
+  end
+
+  def test_post_invoices_generate_pdf
+    # There is currently a bug with openApi generation where endpoints that return File type are executed twice.
+    # [invoices_api.rb:103] calls the client with the method `call_api`.
+    # In that method, the tempfile is generated from an api call if the return type is File by executing the request in the `download_file` method, but it also directly calls `request.run`
+
+    request, response = TestUtils.generate_payloads_from('invoicingapi/invoices_generate_pdf_post')
+    TestUtils.setup_expectation(request, response, 2)
```
Comment (fybJenkins) [inline tests/tc_invoicing_api.rb:116]: Could we add a comment here to explain why we're checking for 2 expectations?

File: tests/tc_payments_api.rb [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+    TestUtils.reset_expectations
+  end
+
+  def test_get_transactions
+    # Setting up expectation
+    request, response = TestUtils.generate_payloads_from('paymentsapi/transactions_get')
+    expectation = TestUtils.setup_expectation(request, response, 1)
+
+    api_instance = PaymentsApi::TransactionsApi.new
+    opts = {
```
Comment (fybJenkins) [inline tests/tc_payments_api.rb]: same here for `TestUtils.generateQueryParams`

File: tests/tc_invoicing_api.rb [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+  def teardown
+    TestUtils.reset_expectations
+  end
+
+  def test_get_invoices
+    request, response = TestUtils.generate_payloads_from('invoicingapi/invoices_get')
+    expectation = TestUtils.setup_expectation(request, response, 1)
+
+    api_instance = InvoicingApi::InvoicesApi.new
+    opts = {
```
Comment (fybJenkins) [inline tests/tc_invoicing_api.rb]: same here for `TestUtils.generateQueryParams`

File: tests/tc_billing_api.rb (hunk 2 above)
Comment (yandrickg) [reply tests/tc_billing_api.rb:60]: The utils method extracts a single list item as its own string, so instead of sending a list of 1 item, it sends only the 1 item. The issue here is that `product_availability_get` is expecting certain opt properties to be lists, so I have to generate it manually. 

I couldn't add a second value to the list in the payload because in the case of `product_category` and `solution`, there is only 1 valid value.

File: .github/workflows/validate-all.yml (hunk 1 above)
Comment (yandrickg) [reply .github/workflows/validate-all.yml:50]: yes, it was removed for testing but forgot to add it back, added in new commit.

File: tests/tc_invoicing_api.rb (hunk 3 above)
Comment (yandrickg) [reply tests/tc_invoicing_api.rb:116]: added some more specific comments on top

File: tests/tc_payments_api.rb (hunk 4 above)
Comment (yandrickg) [reply tests/tc_payments_api.rb]: another issue with the extraction is that the opt key name will be in camelCase, while the api method is expecting the key names to be in snake case. I could have extracted ops and then added/fixed the camelCase offenders, but I thought that would be more confusing in the long run.

File: tests/tc_invoicing_api.rb (hunk 5 above)
Comment (yandrickg) [reply tests/tc_invoicing_api.rb]: same issue as in payments

File: tests/tc_billing_api.rb (hunk 2 above)
Comment (fybJenkins) [reply tests/tc_billing_api.rb:60]: Fair enough. Although we could technically update the utils, might not be worth it for edge cases like this considering we only have one API afaik with list-query-params.

File: tests/tc_invoicing_api.rb (hunk 3 above)
Comment (fybJenkins) [reply tests/tc_invoicing_api.rb:116]: Ah, didn't read it - could be moved directly over the `setup_expectation` but this should be fine either way imo.

File: tests/tc_payments_api.rb (hunk 4 above)
Comment (fybJenkins) [reply tests/tc_payments_api.rb]: It might still be better to extract and fix offenders, because this method is used elsewhere and I think not using it at all for individual offenders is more confusing...

File: tests/tc_invoicing_api.rb (hunk 5 above)
Comment (fybJenkins) [reply tests/tc_invoicing_api.rb]: Same here

File: tests/tc_payments_api.rb (hunk 4 above)
Comment (yandrickg) [reply tests/tc_payments_api.rb]: done, another issue popped up where the opts are wanted in integer type, but mockserver is expecting expectations to be setup with string values, fixed that by converting the offending opts pairs into integers manually.

File: tests/tc_invoicing_api.rb (hunk 5 above)
Comment (yandrickg) [reply tests/tc_invoicing_api.rb]: done, found same issue as with payments
